src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/FlowAnnotations.cs

```diff
@@ -592,41 +592,18 @@ protected override TypeAnnotations CreateValueFromKey(TypeDesc key)
                     {
-                        FieldDesc? backingField = null;
-                        if ((property.SetMethod is not null
-                                && property.SetMethod.HasCustomAttribute("System.Runtime.CompilerServices", "CompilerGeneratedAttribute")
-                                && backingFieldFromSetter is null)
-                            || (property.GetMethod is not null
-                                && property.GetMethod.HasCustomAttribute("System.Runtime.CompilerServices", "CompilerGeneratedAttribute")
-                                && backingFieldFromGetter is null))
+                        if (annotatedFields.Any(a => a.Field == backingField))
                         {
-                            // We failed to find the backing field of an auto-property accessor
-                            _logger.LogWarning(property, DiagnosticId.DynamicallyAccessedMembersCouldNotFindBackingField, property.GetDisplayName());
+                            _logger.LogWarning(backingField, DiagnosticId.DynamicallyAccessedMembersOnPropertyConflictsWithBackingField, property.GetDisplayName(), backingField.GetDisplayName());
                         }
-                        else if (backingFieldFromGetter is not null && backingFieldFromSetter is not null
-                            && backingFieldFromSetter != backingFieldFromGetter)
+                        else if (!(backingFieldFromGetter is not null
+                            && backingFieldFromSetter is not null
+                            && backingFieldFromGetter != backingFieldFromSetter))
                         {
-                            // We found two different backing fields for the getter and the setter
-                            _logger.LogWarning(property, DiagnosticId.DynamicallyAccessedMembersCouldNotFindBackingField, property.GetDisplayName());
-                        }
-                        else
-                        {
-                            // We either have a single auto-property accessor or both accessors point to the same backing field
-                            backingField = backingFieldFromSetter ?? backingFieldFromGetter;
-                        }
-
-                        if (backingField != null)
-                        {
-                            if (annotatedFields.Any(a => a.Field == backingField))
-                            {
-                                _logger.LogWarning(backingField, DiagnosticId.DynamicallyAccessedMembersOnPropertyConflictsWithBackingField, property.GetDisplayName(), backingField.GetDisplayName());
-                            }
-                            else
-                            {
-                                annotatedFields.Add(new FieldAnnotation(backingField, annotation));
-                            }
+                            annotatedFields.Add(new FieldAnnotation(backingField, annotation));
                         }
```

Comment:
It might be worth a comment here to explain that we're in a situation where we either didn't find a back field or found two backing fields, and we're ignoring it.